File: Serenity Practice/HosbitalSerenity.Web/Modules/Default/Governorate/GovernorateRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace HosbitalSerenity.Default;

[ConnectionKey("Default"), Module("Default"), TableName("Governorate")]
[DisplayName("Governorate"), InstanceName("Governorate")]
[ReadPermission("?")]
[ModifyPermission("?")]
[DeletePermission("?")]
[NavigationPermission("Administration:General")]
[ServiceLookupPermission("?")]
[LookupScript("Default.Governorate")]
public sealed class GovernorateRow : Row<GovernorateRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Gov Id"), Column("govId"), Identity, IdProperty]
    public int? GovId { get => fields.GovId[this]; set => fields.GovId[this] = value; }

    [DisplayName("Gov Name"), Column("govName"), Size(100), NotNull, QuickSearch, NameProperty]
    public string GovName { get => fields.GovName[this]; set => fields.GovName[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field GovId;
        public StringField GovName;

    }
}

File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Default/GovernorateRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface GovernorateRow {
    GovId?: number;
    GovName?: string;
}

export abstract class GovernorateRow {
    static readonly idProperty = 'GovId';
    static readonly nameProperty = 'GovName';
    static readonly localTextPrefix = 'Default.Governorate';
    static readonly lookupKey = 'Default.Governorate';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<GovernorateRow>('Default.Governorate') }
    static async getLookupAsync() { return getLookupAsync<GovernorateRow>('Default.Governorate') }

    static readonly deletePermission = '?';
    static readonly insertPermission = '?';
    static readonly readPermission = '?';
    static readonly updatePermission = '?';

    static readonly Fields = fieldsProxy<GovernorateRow>();
}